salt/client/__init__.py

```diff
@@ -1684,6 +1684,8 @@ def pub(self,
             )
             raise SaltClientError
 
+        self._check_cmd_avail(fun, tgt, expr_form, arg)
```

Comment:
`expr_form` has been deprecated and removed. This should be using `tgt_type` instead now.